### DISCUSSION TITLE
please help me write def file

### DISCUSSION THREAD

**Discussion by daipvd** [RTimothyEdwards/magic#320] [General]
Hi Everyone,
This is first time I use magic tool and 
I'm trying to write .def file from "tut3e" example, here is steps I did:
1. Open tut3e on magic tool
2. on command window of magic I type "extract" command
3. type "def write test.def"
But there is no .def was writed. And here is my run log:
![image](https://github.com/user-attachments/assets/a8839467-8546-49f5-909f-ecb434450da1)
Can you show me what am I wrong here?

Thank you a lot,
dai

**Comment by RTimothyEdwards** [comment]
What version of magic?

I did what you did and I get a "tut3e.def" file, and every indication from the output is that you got the same behavior.

Now, I don't get a very _useful_ DEF file, mainly because DEF was designed for standard cell layouts and doesn't have any way of describing transistors.  Also, the tutorial examples are in the SCMOS technology which does not have a "lef" section and therefore doesn't know how to write the output layers, and so it produces a DEF file with section headers but no content.
